openedx_events/learning/data.py

```diff
@@ -0,0 +1,137 @@
+"""
+Data attributes for events within the architecture subdomain `learning`.
+
+These attributes follow the form of attr objects specified in OEP-49 data
+pattern.
+"""
+from datetime import datetime
+
+import attr
+from opaque_keys.edx.keys import CourseKey
+
+
+@attr.s(frozen=True)
+class UserNonPersonalData:
+    """
+    Attributes defined for Open edX user object based on non-PII data.
+
+    Arguments:
+        id (int): unique identifier for the Django User object.
+        is_active (bool): indicates whether the user is active.
+    """
+
+    id = attr.ib(type=int)
+    is_active = attr.ib(type=bool)
+
+
+@attr.s(frozen=True)
+class UserPersonalData:
+    """
+    Attributes defined for Open edX user object based on PII data.
+
+    Arguments:
+        username (str): username associated with the Open edX user.
+        email (str): email associated with the Open edX user.
+        name (str): email associated with the Open edX user's profile.
+    """
+
+    username = attr.ib(type=str)
+    email = attr.ib(type=str)
+    name = attr.ib(type=str, factory=str)
+
+
+@attr.s(frozen=True)
+class UserData(UserNonPersonalData):
```

Comment:
Hey @felipemontoya, thanks for the recommendation you did here.

Reply:
I think we are